File: Practice/06/C++/6. Корни уравнения/6. Корни уравнения/6. Корни уравнения.cpp

#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    double a, b, c, x;
    setlocale(LC_ALL, "Russian");
    cout << "Введите данные для уравнения типа ax^2+bx+c=0" << endl << "a=";
    cin >> a;
    cout << "b=";
    cin >> b;
    cout << "c=";
    cin >> c;
    double D = b * b - 4 * a * c;
    if (a == 0) {
        x = -c / b;
        cout << "x=" << x << endl;
    }
    else {
        if (D > 0) {
            x = (-1 * b + sqrt(D)) / (2 * a);
            cout << "x1=" << x << endl;
            x = (-1 * b - sqrt(D)) / (2 * a);
            cout << "x2=" << x << endl;
        }
        else if (D == 0) {
            x = (-1 * b + sqrt(D)) / (2 * a);
            cout << "x=" << x << endl;
        }
        else {
            cout << "уравнение не имеет корней" << endl;
        }
   }
}

File: Practice/06/Python/6. Корни уравнения/6. Корни уравнения/_6._Корни_уравнения.py
import math
print ("Введите данные для уравнения ax^2+bx+c=0")

a=float(input("a="))
b=float(input("b="))
c=float(input("c="))

D = (b * b - (4 * a * c))
if a==0:
    x=(-c/b)
    print("x=",x)
elif D>0:
    x1=(-b+math.sqrt(D))/(2*a)
    print("x1=",x1)
    x2=(-b-math.sqrt(D))/(2*a)
    print("x2=",x2)
elif D==0:
    x=-b/(2*a)
    print("x=",x)
else:
    print ("Корней нет")

